code/game/turfs/simulated/water.dm

```diff
@@ -108,11 +115,109 @@
 		return species.can_breathe_water()
 	return ..()
 
-// Use this to have things react to having water applied to them.
-/atom/movable/proc/water_act(amount)
-	return
-
-/mob/living/water_act(amount)
-	adjust_fire_stacks(-amount * 5)
-	for(var/atom/movable/AM in contents)
-		AM.water_act(amount)
\ No newline at end of file
+/turf/simulated/floor/beach/water/proc/wash_floor(var/turf/T)
+	T.clean(src)
```

Comment:
This is where your runtime error is being caused from. It's a little weird as you're basically calling
src.clean(src) sometimes and then sometimes null.clean(src)